src/main/handlers/status_check.py

```diff
@@ -1,45 +1,54 @@
-import requests
+import socket
 import asyncio
 
 from datetime import datetime
 
 import discord
-
-
-def get_status(data):
-    try:
-        map_name = data['map']
-        playercount = data['playerscount'][:data['playerscount'].index('/')]
-        game_str = f'{map_name} | {playercount} player' + 's' if playercount != '1' else ''
-
-        print(datetime.now().isoformat(), game_str)
-        return discord.Status.online, discord.Game(name=game_str)
-
-    except KeyError as error:
-        print(error, data)
-        if data['playerscount'] == 'offline':
-            return discord.Status.dnd, None
-
-    except BaseException as error:
-        print(error, data)
-        return discord.Status.invisible, None
+import steam.game_servers
 
 
 class StatusCheck:
 
-    def __init__(self, client):
+    def __init__(self, client, config):
         self.client = client
+        self.config = config
+
+        self.failed_checks = 0
+        self.maintenance = False
 
     async def execute(self):
-        # TODO stop using while True
-        while True:
-            print('\nChecking status...')
-
-            try:
-                data = requests.get('https://api.trackyserver.com/widget/index.php?id=300852').json()
-                status, activity = get_status(data)
-                await self.client.change_presence(status=status, activity=activity)
-            except BaseException as error:
-                print(datetime.now().isoformat(), error)
-            finally:
-                await asyncio.sleep(60)
+        try:
+            status, activity = self.get_status()
+            await self.client.change_presence(status=status, activity=activity)
+            print(f'{datetime.now().isoformat()} {status} - {activity}')
+        finally:
+            await asyncio.sleep(30)
+            self.client.loop.create_task(self.execute())
+
+    def get_status(self):
+        print('\nChecking status...')
+        try:
+            data = steam.game_servers.a2s_info(('147.135.85.168', 27015))
+
+            if data['visibility']:
+                self.maintenance = True
+                return discord.Status.dnd, 'maintenance'
+            else:
+                game_str = f'{data["map"]} | {data["players"]} player' + ('s' if data["players"] != '1' else '')
+                self.maintenance = False
+                self.failed_checks = 0
+                return discord.Status.online, discord.Game(name=game_str)
+
+        except socket.timeout as error:
+            print(error, f'Failed checks: {self.failed_checks}')
+            self.failed_checks += 1
+            if self.failed_checks == 4 and not self.maintenance:
+                self.client.loop.create_task(self.alert_sleg())
```

Comment:
Ditto. https://github.com/MarkoDedovic/discord-bot/pull/5/files#diff-36a2e40a3fac2ba52a447e428d7cd3f7cd559b3797e6f25c198869cb6c0eea92R21
```suggestion
                await self.alert_sleg()
```

Reply:
alert_sleg() is an async method so it can't be directly called from a synchronous method

self.client.loop accesses the discord client and its event loop so we can toss on another task with loop.create_task()